### ISSUE TITLE
Issue about father-component's scoped styles when child-components shows by v-if

### ISSUE THREAD

**Issue by chairuosen** [vuejs/vue#4915]
### Vue.js version
2.1.10 // the latest

### Reproduction Link

https://github.com/chairuosen/vue-bug-demo

Expected in this demo : all the text is red.
Got: only 1 and 3 is red.

### Steps to reproduce

1. create a father component that has a class `item` on each of his child.
```vue
<style scoped>
  .item{
    color:red;
  }
</style>
<template>
     <child ref="child" class="item"></child>
</template>
```

2. create the child
```vue
<template>
  <div v-if="active">
      Im the child
  </div>
</template>

<script>
  export default {
    data () {
      return {
        active:false
      }
    },
    methods:{
      show:function () {
        this.active = true;
      }
    }
  }
</script>
```

3. the default state of the child is hidden.
show it like `this.$refs.child.show();` or others.

### What is Expected?

Expected : Text ` Im the child` with red color

### What is actually happening?

black.

### Probably Reason

vue will add some private attribute to all the element of the component when it has scoped styles.
but forgot to do it with the `v-if=false` default hidden child

**Comment by LinusBorg** [comment]
I would rate this a bug. Thanks.

**Comment by yyx990803** [comment]
This is already fixed in 90a455c95ceb5d38e5a26b172e958232c25265d2
